programs/server/Server.cpp

```diff
@@ -781,9 +790,12 @@ void loadStartupScripts(const Poco::Util::AbstractConfiguration & config, Contex
             startup_context->makeQueryContext();
             executeQuery(read_buffer, write_buffer, true, startup_context, callback, QueryFlags{ .internal = true }, std::nullopt, {});
         }
+
```

Comment:
I think it's potentially misleading. Server startup can get "stuck" before `loadStartupScripts` is executed and then it will be unclear from the code what state the metric should be in. This is why it's intentionally not set to `StartupScriptsExecutionState::NotFinished` anywhere in the code.